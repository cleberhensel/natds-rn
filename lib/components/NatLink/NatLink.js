import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text } from './NatLink.styles';

export const NatLink = ({
  value, onPress, type, margin,
}) => {
  const defaultMargin = (propMargin) => {
    switch (propMargin) {
      case 'none': return '0px';
      case 'dense': return '8px';
      case 'normal': return '16px';
      default: return '0px';
    }
  };

  return (
    <TouchableOpacity onPress={onPress} defaultMargin={defaultMargin(margin)}>
      <Text type={type}>
        {value}
      </Text>
    </TouchableOpacity>
  );
};

NatLink.propTypes = {
  margin: PropTypes.string,
  onPress: PropTypes.func,
  type: PropTypes.string,
  value: PropTypes.string,
};

NatLink.defaultProps = {
  type: 'body2',
};
