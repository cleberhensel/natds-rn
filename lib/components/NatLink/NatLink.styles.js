import styled from 'styled-components/native';
import { colors } from '../../assets/theme/colors/colors';

const linkStyles = {
  body2: {
    fontSize: 14,
    color: colors.bw.grayDark,
    lineHeight: 21,
    fontWeight: 400,
  },
};

const getStylesFrom = styles => type => styles[type] || styles.default;

const getLinkStyles = getStylesFrom(linkStyles);

export const TouchableOpacity = styled.TouchableOpacity`
  flex: 1;
  margin-top: ${({ defaultMargin }) => defaultMargin || '0px'};
`;

export const Text = styled.Text`
  font-size: ${({ type }) => getLinkStyles(type).fontSize};
  color: ${({ type }) => getLinkStyles(type).color};
  line-height: ${({ type }) => getLinkStyles(type).lineHeight};
  font-weight: ${({ type }) => getLinkStyles(type).fontWeight};
`;
