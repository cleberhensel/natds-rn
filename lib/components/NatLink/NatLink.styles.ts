import styled from "styled-components/native";
import { sizes } from "../../tokens/sizes";
export const TouchableOpacity = styled.TouchableOpacity`
  flex: 1;
  margin-bottom: ${({ marginBottom }) => sizes[marginBottom]}px;
  align-items: ${({ alignDefault }) => alignDefault};
`;
