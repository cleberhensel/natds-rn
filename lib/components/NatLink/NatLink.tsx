import React from 'react';
import { TouchableOpacity } from './NatLink.styles';
import { ThemeNaturaLight as Theme } from '../../themes/ThemeNaturaLight';
import { NatText } from '../NatText';

type NatLinkProps = {
  align?: string,
  color?: string,
  margin?: string,
  onPress?: (...args: any[]) => any,
  type?: string,
  value?: string
};
export const NatLink: React.SFC<NatLinkProps> = ({
  value,
  onPress,
  type,
  margin,
  color,
  align,
}) => (
  <TouchableOpacity
    onPress={onPress}
    marginBottom={margin}
    alignDefault={align}
  >
    <NatText type={type} value={value} color={color} />
  </TouchableOpacity>
);
NatLink.defaultProps = {
  align: 'flex-start',
  color: Theme.palette.complementary.link,
  margin: 'none',
  type: 'body2',
};
