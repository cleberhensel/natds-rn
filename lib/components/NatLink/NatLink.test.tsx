import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';
import { NatLink } from './NatLink';

describe('NatText component default', () => {
  it('Should render NatText component ', () => {
    const button = renderer.create(<NatLink value="Teste" />).toJSON();

    expect(button).toMatchSnapshot('NatLink snapshot');
  });
});
