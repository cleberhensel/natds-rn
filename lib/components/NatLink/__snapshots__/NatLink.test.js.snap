// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatText component default Should render NatText component  1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": 0,
      "opacity": 1,
    }
  }
>
  <View
    margin={24}
    style={
      Array [
        Object {
          "marginBottom": 24,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Text
      color="#227BBD"
      style={
        Array [
          Object {
            "color": "#227BBD",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 21,
          },
        ]
      }
      type="body2"
    >
      Teste
    </Text>
  </View>
</View>
`;
